tests/FakeItEasy.Tests/Expressions/ExpressionArgumentConstraintFactoryTests.cs

```diff
@@ -44,6 +44,20 @@ public void Should_return_constraint_from_trapper_when_available()
             result.Should().BeSameAs(constraint);
         }
 
+
+        [Test]
+        public void Should_return_null_from_trapper_when_set_to_return_null()

```

Comment:
I don't think you need to add this test. The test above isn't a test for `WithReturnType`, but a test for `ExpressionArgumentConstraintFactory`.